src/cmd/sql_cmd.h

```diff
@@ -1463,18 +122,19 @@ void Shell() {
         ns_endpoint = ns_client->GetEndpoint();
     }
 
-    std::string display_prefix = ns_endpoint + "/" + db + "> ";
+    std::string display_prefix = ns_endpoint + "/" + sr->GetDatabase() + "> ";
     std::string multi_line_perfix = std::string(display_prefix.length() - 3, ' ') + "-> ";
     std::string sql;
     bool multi_line = false;
     while (true) {
         std::string buffer;
         if (!FLAGS_interactive) {
             buffer = FLAGS_cmd;
-            db = FLAGS_database;
+            std::string db = FLAGS_database;
```

Comment:
How about `if(!FLAGS_cmd.empty){buffer=FLAGS_cmd;}`? `FLAG_interactive` is only  a config.
So we can `echo "drop table xx;" | openmldb --sql_client --interactive=false`.